lib/doc/worksheet.js

```diff
@@ -579,6 +584,29 @@ class Worksheet {
     return image && image.imageId;
   }
 
+  // =========================================================================
+  // Worksheet Protection
+  protect(password, options){
+    this.sheetProtection.sheet = true;
+    if (password) {
+        this.sheetProtection.algorithmName = 'SHA-512';
+        this.sheetProtection.saltValue = Encryptor.randomBytes(16).toString('base64');
+        this.sheetProtection.spinCount = 100000;
+        this.sheetProtection.hashValue = Encryptor.convertPasswordToHash(password, 'SHA512', this.sheetProtection.saltValue, this.sheetProtection.spinCount);
+    }
+    if(options){
+      this.sheetProtection = Object.assign(this.sheetProtection, options);
+    }
+  }
+
+  unprotect(){
```

Comment:
This can just assign sheetProtection to null

Reply:
Wrapped this into a promise and changed commit() in worksheet-writer. I cannot determine any improvements.

Reply:
Cool - it's not really about improvements but future proofing the API. In the case that one day we might be able to make this call properly asynchronous. 700 milliseconds is a very long time to be stuck adding a password hash to a sheet and if we can pass that over to a worker thread it will be much better for the app as a whole.